Compilation issue v131

I’m able to compile 130.1 with no issues via:
make -j5 hiro=gtk3

In v131 I get:
Compiling ares/n64/cpu/cpu.cpp ...
Compiling ares/n64/vulkan/parallel-rdp/parallel-rdp/command_ring.cpp ...
Compiling ares/n64/vulkan/parallel-rdp/parallel-rdp/rdp_device.cpp ...
Compiling ares/n64/vulkan/parallel-rdp/parallel-rdp/rdp_dump_write.cpp ...
Compiling ares/n64/vulkan/parallel-rdp/parallel-rdp/rdp_renderer.cpp ...
Compiling ares/n64/vulkan/parallel-rdp/parallel-rdp/video_interface.cpp ...
Compiling ares/n64/vulkan/parallel-rdp/vulkan/buffer.cpp ...
Compiling ares/n64/vulkan/parallel-rdp/vulkan/buffer_pool.cpp ...
Compiling ares/n64/vulkan/parallel-rdp/vulkan/command_buffer.cpp ...
Compiling ares/n64/vulkan/parallel-rdp/vulkan/command_pool.cpp ...
Compiling ares/n64/vulkan/parallel-rdp/vulkan/context.cpp ...
Compiling ares/n64/vulkan/parallel-rdp/vulkan/cookie.cpp ...
Compiling ares/n64/vulkan/parallel-rdp/vulkan/descriptor_set.cpp ...
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:32:7: error: '__builtin_ia32_roundss' needs target feature sse4.1
  t = _mm_round_ss(t, t, _MM_FROUND_TO_POS_INF);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:281:12: note: expanded from macro '_mm_round_ss'
  ((__m128)__builtin_ia32_roundss((__v4sf)(__m128)(X), (__v4sf)(__m128)(Y),    \
           ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:43:7: error: '__builtin_ia32_roundsd' needs target feature sse4.1
  t = _mm_round_sd(t, t, _MM_FROUND_TO_POS_INF);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:356:13: note: expanded from macro '_mm_round_sd'
  ((__m128d)__builtin_ia32_roundsd((__v2df)(__m128d)(X), (__v2df)(__m128d)(Y), \
            ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:32:7: error: '__builtin_ia32_roundss' needs target feature sse4.1
  t = _mm_round_ss(t, t, _MM_FROUND_TO_POS_INF);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:281:12: note: expanded from macro '_mm_round_ss'
  ((__m128)__builtin_ia32_roundss((__v4sf)(__m128)(X), (__v4sf)(__m128)(Y),    \
           ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:43:7: error: '__builtin_ia32_roundsd' needs target feature sse4.1
  t = _mm_round_sd(t, t, _MM_FROUND_TO_POS_INF);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:356:13: note: expanded from macro '_mm_round_sd'
  ((__m128d)__builtin_ia32_roundsd((__v2df)(__m128d)(X), (__v2df)(__m128d)(Y), \
            ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:54:7: error: '__builtin_ia32_roundss' needs target feature sse4.1
  t = _mm_round_ss(t, t, _MM_FROUND_CUR_DIRECTION);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:281:12: note: expanded from macro '_mm_round_ss'
  ((__m128)__builtin_ia32_roundss((__v4sf)(__m128)(X), (__v4sf)(__m128)(Y),    \
           ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:65:7: error: '__builtin_ia32_roundsd' needs target feature sse4.1
  t = _mm_round_sd(t, t, _MM_FROUND_CUR_DIRECTION);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:356:13: note: expanded from macro '_mm_round_sd'
  ((__m128d)__builtin_ia32_roundsd((__v2df)(__m128d)(X), (__v2df)(__m128d)(Y), \
            ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:54:7: error: '__builtin_ia32_roundss' needs target feature sse4.1
  t = _mm_round_ss(t, t, _MM_FROUND_CUR_DIRECTION);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:281:12: note: expanded from macro '_mm_round_ss'
  ((__m128)__builtin_ia32_roundss((__v4sf)(__m128)(X), (__v4sf)(__m128)(Y),    \
           ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:65:7: error: '__builtin_ia32_roundsd' needs target feature sse4.1
  t = _mm_round_sd(t, t, _MM_FROUND_CUR_DIRECTION);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:356:13: note: expanded from macro '_mm_round_sd'
  ((__m128d)__builtin_ia32_roundsd((__v2df)(__m128d)(X), (__v2df)(__m128d)(Y), \
            ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:76:7: error: '__builtin_ia32_roundss' needs target feature sse4.1
  t = _mm_round_ss(t, t, _MM_FROUND_TO_NEG_INF);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:281:12: note: expanded from macro '_mm_round_ss'
  ((__m128)__builtin_ia32_roundss((__v4sf)(__m128)(X), (__v4sf)(__m128)(Y),    \
           ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:87:7: error: '__builtin_ia32_roundsd' needs target feature sse4.1
  t = _mm_round_sd(t, t, _MM_FROUND_TO_NEG_INF);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:356:13: note: expanded from macro '_mm_round_sd'
  ((__m128d)__builtin_ia32_roundsd((__v2df)(__m128d)(X), (__v2df)(__m128d)(Y), \
            ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:76:7: error: '__builtin_ia32_roundss' needs target feature sse4.1
  t = _mm_round_ss(t, t, _MM_FROUND_TO_NEG_INF);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:281:12: note: expanded from macro '_mm_round_ss'
  ((__m128)__builtin_ia32_roundss((__v4sf)(__m128)(X), (__v4sf)(__m128)(Y),    \
           ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:87:7: error: '__builtin_ia32_roundsd' needs target feature sse4.1
  t = _mm_round_sd(t, t, _MM_FROUND_TO_NEG_INF);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:356:13: note: expanded from macro '_mm_round_sd'
  ((__m128d)__builtin_ia32_roundsd((__v2df)(__m128d)(X), (__v2df)(__m128d)(Y), \
            ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:7:7: error: '__builtin_ia32_roundss' needs target feature sse4.1
  t = _mm_round_ss(t, t, _MM_FROUND_TO_NEAREST_INT);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:281:12: note: expanded from macro '_mm_round_ss'
  ((__m128)__builtin_ia32_roundss((__v4sf)(__m128)(X), (__v4sf)(__m128)(Y),    \
           ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:21:7: error: '__builtin_ia32_roundsd' needs target feature sse4.1
  t = _mm_round_sd(t, t, _MM_FROUND_TO_NEAREST_INT);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:356:13: note: expanded from macro '_mm_round_sd'
  ((__m128d)__builtin_ia32_roundsd((__v2df)(__m128d)(X), (__v2df)(__m128d)(Y), \
            ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:7:7: error: '__builtin_ia32_roundss' needs target feature sse4.1
  t = _mm_round_ss(t, t, _MM_FROUND_TO_NEAREST_INT);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:281:12: note: expanded from macro '_mm_round_ss'
  ((__m128)__builtin_ia32_roundss((__v4sf)(__m128)(X), (__v4sf)(__m128)(Y),    \
           ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:21:7: error: '__builtin_ia32_roundsd' needs target feature sse4.1
  t = _mm_round_sd(t, t, _MM_FROUND_TO_NEAREST_INT);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:356:13: note: expanded from macro '_mm_round_sd'
  ((__m128d)__builtin_ia32_roundsd((__v2df)(__m128d)(X), (__v2df)(__m128d)(Y), \
            ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:98:7: error: '__builtin_ia32_roundss' needs target feature sse4.1
  t = _mm_round_ss(t, t, _MM_FROUND_TO_ZERO);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:281:12: note: expanded from macro '_mm_round_ss'
  ((__m128)__builtin_ia32_roundss((__v4sf)(__m128)(X), (__v4sf)(__m128)(Y),    \
           ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:109:7: error: '__builtin_ia32_roundsd' needs target feature sse4.1
  t = _mm_round_sd(t, t, _MM_FROUND_TO_ZERO);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:356:13: note: expanded from macro '_mm_round_sd'
  ((__m128d)__builtin_ia32_roundsd((__v2df)(__m128d)(X), (__v2df)(__m128d)(Y), \
            ^
In file included from ../ares/n64/cpu/cpu.cpp:11:
../ares/n64/cpu/algorithms.cpp:98:7: error: '__builtin_ia32_roundss' needs target feature sse4.1
  t = _mm_round_ss(t, t, _MM_FROUND_TO_ZERO);
      ^
/usr/lib/clang/15.0.6/include/smmintrin.h:281:12: note: expanded from macro '_mm_round_ss'
  ((__m128)__builtin_ia32_roundss((__v4sf)(__m128)(X), (__v4sf)(__m128)(Y),    \
           ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
Compiling ares/n64/vulkan/parallel-rdp/vulkan/device.cpp ...
20 errors generated.
make[1]: *** [../nall/GNUmakefile:265: obj/ares-n64-cpu.o] Error 1

--- a/ares/n64/cpu/algorithms.cpp
+++ b/ares/n64/cpu/algorithms.cpp
@@ -2,7 +2,7 @@ template <typename T>
 auto CPU::roundNearest(f32 f) -> T {
 #if defined(ARCHITECTURE_ARM64)
   return vrndns_f32(f);
-#elif defined(ARCHITECTURE_AMD64)
+#elif defined(ARCHITECTURE_AMD64) && ARCHITECTURE_SUPPORTS_SSE4_1
   __m128 t = _mm_set_ss(f);
   t = _mm_round_ss(t, t, _MM_FROUND_TO_NEAREST_INT);
   return _mm_cvtss_f32(t);
@@ -16,7 +16,7 @@ auto CPU::roundNearest(f64 f) -> T {
 #if defined(ARCHITECTURE_ARM64)
   float64x1_t vf = {f};
   return vrndn_f64(vf)[0];
-#elif defined(ARCHITECTURE_AMD64)
+#elif defined(ARCHITECTURE_AMD64) && ARCHITECTURE_SUPPORTS_SSE4_1
   __m128d t = _mm_set_sd(f);
   t = _mm_round_sd(t, t, _MM_FROUND_TO_NEAREST_INT);
   return _mm_cvtsd_f64(t);
@@ -27,7 +27,7 @@ auto CPU::roundNearest(f64 f) -> T {
 
 template <typename T>
 auto CPU::roundCeil(f32 f) -> T {
-#if defined(ARCHITECTURE_AMD64)
+#if defined(ARCHITECTURE_AMD64) && ARCHITECTURE_SUPPORTS_SSE4_1
   __m128 t = _mm_set_ss(f);
   t = _mm_round_ss(t, t, _MM_FROUND_TO_POS_INF);
   return _mm_cvtss_f32(t);
@@ -38,7 +38,7 @@ auto CPU::roundCeil(f32 f) -> T {
 
 template <typename T>
 auto CPU::roundCeil(f64 f) -> T {
-#if defined(ARCHITECTURE_AMD64)
+#if defined(ARCHITECTURE_AMD64) && ARCHITECTURE_SUPPORTS_SSE4_1
   __m128d t = _mm_set_sd(f);
   t = _mm_round_sd(t, t, _MM_FROUND_TO_POS_INF);
   return _mm_cvtsd_f64(t);
@@ -49,7 +49,7 @@ auto CPU::roundCeil(f64 f) -> T {
 
 template<typename T>
 auto CPU::roundCurrent(f32 f) -> T {
-#if defined(ARCHITECTURE_AMD64)
+#if defined(ARCHITECTURE_AMD64) && ARCHITECTURE_SUPPORTS_SSE4_1
   auto t = _mm_set_ss(f);
   t = _mm_round_ss(t, t, _MM_FROUND_CUR_DIRECTION);
   return _mm_cvtss_f32(t);
@@ -60,7 +60,7 @@ auto CPU::roundCurrent(f32 f) -> T {
 
 template<typename T>
 auto CPU::roundCurrent(f64 f) -> T {
-#if defined(ARCHITECTURE_AMD64)
+#if defined(ARCHITECTURE_AMD64) && ARCHITECTURE_SUPPORTS_SSE4_1
   auto t = _mm_set_sd(f);
   t = _mm_round_sd(t, t, _MM_FROUND_CUR_DIRECTION);
   return _mm_cvtsd_f64(t);
@@ -71,7 +71,7 @@ auto CPU::roundCurrent(f64 f) -> T {
 
 template <typename T>
 auto CPU::roundFloor(f32 f) -> T {
-#if defined(ARCHITECTURE_AMD64)
+#if defined(ARCHITECTURE_AMD64) && ARCHITECTURE_SUPPORTS_SSE4_1
   __m128 t = _mm_set_ss(f);
   t = _mm_round_ss(t, t, _MM_FROUND_TO_NEG_INF);
   return _mm_cvtss_f32(t);
@@ -82,7 +82,7 @@ auto CPU::roundFloor(f32 f) -> T {
 
 template <typename T>
 auto CPU::roundFloor(f64 f) -> T {
-#if defined(ARCHITECTURE_AMD64)
+#if defined(ARCHITECTURE_AMD64) && ARCHITECTURE_SUPPORTS_SSE4_1
   __m128d t = _mm_set_sd(f);
   t = _mm_round_sd(t, t, _MM_FROUND_TO_NEG_INF);
   return _mm_cvtsd_f64(t);
@@ -93,7 +93,7 @@ auto CPU::roundFloor(f64 f) -> T {
 
 template <typename T>
 auto CPU::roundTrunc(f32 f) -> T {
-#if defined(ARCHITECTURE_AMD64)
+#if defined(ARCHITECTURE_AMD64) && ARCHITECTURE_SUPPORTS_SSE4_1
   __m128 t = _mm_set_ss(f);
   t = _mm_round_ss(t, t, _MM_FROUND_TO_ZERO);
   return _mm_cvtss_f32(t);
@@ -104,7 +104,7 @@ auto CPU::roundTrunc(f32 f) -> T {
 
 template <typename T>
 auto CPU::roundTrunc(f64 f) -> T {
-#if defined(ARCHITECTURE_AMD64)
+#if defined(ARCHITECTURE_AMD64) && ARCHITECTURE_SUPPORTS_SSE4_1
   __m128d t = _mm_set_sd(f);
   t = _mm_round_sd(t, t, _MM_FROUND_TO_ZERO);
   return _mm_cvtsd_f64(t);

